catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.java

```diff
@@ -123,6 +127,21 @@ public boolean onOptionsItemSelected(MenuItem item) {
 		return super.onOptionsItemSelected(item);
 	}
 
+	private void checkIfCrashRecoveryAndFinishActivity(final Activity context) {
+		if (isRecoveredFromCrash()) {
+			if (!(context instanceof MainMenuActivity)) {
+				context.finish();
+			} else {
+				PreferenceManager.getDefaultSharedPreferences(this).edit().putBoolean(RECOVERED_FROM_CRASH, false).commit();
+			}
+		}
+		CrashReporter.sendUnhandledCaughtException();
```

Comment:
Please move the line in the if isRecoveredFromCrash block.
```java
if (isRecoveredFromCrash()) {
  CrashReporter.sendUnhandledCaughtException();
  …
```

This way we don't change the behavior in any kind. We will discuss where to put it later on.